File: typescript/src/3014-minimumPushes.ts
function minimumPushes(word: string): number {
  const len: number = word.length;
  if (len <= 8) {
    return len;
  } else if (len <= 16) {
    return 8 + (len - 8) * 2;
  } else if (len <= 24) {
    return 8 + 8 * 2 + (len - 16) * 3;
  } else {
    return 8 + 8 * 2 + 8 * 3 + (len - 24) * 4;
  }
}

export {minimumPushes};

File: python/src/3014-minimumPushes.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        strLen = len(word)

        if strLen <= 8:
            return strLen

        if (strLen <= 16):
            return 8 + (strLen - 8) * 2
        
        if (strLen <= 24):
            return 8 + 8 * 2 + (strLen - 16) * 3

        return 8 + 8 * 2 + 8 * 3 + (strLen - 24) * 4

